src/test/java/com/puppycrawl/tools/checkstyle/internal/CliOptionsXdocsSyncTest.java

```diff
@@ -0,0 +1,102 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.puppycrawl.tools.checkstyle.internal.utils.XmlUtil;
+import picocli.CommandLine;
+
+public class CliOptionsXdocsSyncTest {
+
+    @Test
+    public void validateCliDocSection() throws Exception {
+        final Map<String, String> cmdDesc = new HashMap<>();
+        final Set<String> ignoreXdocToMain = new HashSet<>();
+        // used to ignore options present in xdocs but not in Main.java
+        ignoreXdocToMain.add("-V");
+        ignoreXdocToMain.add("-h");
+
+        final Path path = Paths.get("src/xdocs/cmdline.xml.vm");
+        final String input = new String(Files.readAllBytes(path), UTF_8);
+
+        final Document document = XmlUtil.getRawXml(path.getFileName().toString(), input, input);
+        final NodeList sections = document.getElementsByTagName("section");
```

Comment:
Please extract this block to a private static helper method.

Reply:
Done